blueprint-manager/src/main/java/com/sequenceiq/cloudbreak/blueprint/kerberos/KerberosDetailService.java

```diff
@@ -19,44 +20,44 @@
 
     private final Gson gson = new Gson();
 
-    public String resolveTypeForKerberos(KerberosConfig kerberosConfig) {
+    public String resolveTypeForKerberos(@Nonnull KerberosConfig kerberosConfig) {
         if (!Strings.isNullOrEmpty(kerberosConfig.getContainerDn()) && !Strings.isNullOrEmpty(kerberosConfig.getLdapUrl())) {
             return "active-directory";
         }
         return "mit-kdc";
     }
 
-    public String resolveHostForKerberos(KerberosConfig kerberosConfig, String defaultHost) {
-        return Strings.isNullOrEmpty(kerberosConfig.getUrl()) ? defaultHost : kerberosConfig.getUrl();
+    public String resolveHostForKerberos(@Nonnull KerberosConfig kerberosConfig, String defaultHost) {
+        return kerberosConfig.getUrl() == null || kerberosConfig.getUrl().trim().isEmpty() ? defaultHost : kerberosConfig.getUrl().trim();
     }
 
-    public String resolveHostForKdcAdmin(KerberosConfig kerberosConfig, String defaultHost) {
-        return Strings.isNullOrEmpty(kerberosConfig.getAdminUrl()) ? defaultHost : kerberosConfig.getAdminUrl();
+    public String resolveHostForKdcAdmin(@Nonnull KerberosConfig kerberosConfig, String defaultHost) {
+        return kerberosConfig.getAdminUrl() == null || kerberosConfig.getAdminUrl().trim().isEmpty() ? defaultHost : kerberosConfig.getAdminUrl().trim();
```

Comment:
Trim happens twice. I prefer something like this:`Optional.ofNullable(kerberosConfig.getAdminUrl()).orElse(").trim()`